src/detect-tls-cert-validity.c

```diff
@@ -1146,14 +1288,13 @@ int ValidityTestDetect01(void)
     FAIL_IF_NOT(PacketAlertCheck(p3, 1));
     FAIL_IF_NOT(PacketAlertCheck(p3, 2));
 
-    if (alp_tctx != NULL)
-        AppLayerParserThreadCtxFree(alp_tctx);
-    if (det_ctx != NULL)
-        DetectEngineThreadCtxDeinit(&tv, det_ctx);
-    if (de_ctx != NULL)
-        SigGroupCleanup(de_ctx);
-    if (de_ctx != NULL)
-        DetectEngineCtxFree(de_ctx);
+    FAIL_IF_NULL(alp_tctx);
+    FAIL_IF_NULL(det_ctx);
```

Comment:
not sure I get the point of checking these here well after they have been dereferenced already